### PR TITLE
Drone dispensers now tell you how much material they need per drone.

### PR DESCRIPTION
**Pull request by NamelessFairy**
## About The Pull Request

Does what the title says, when you inspect a drone dispenser that has a material requirement it tells you what that material requirement is.
## Why It's Good For The Game

Played a round earlier today where I wanted to fill a drone dispenser and I had to work out how many materials it needed by just inserting a boatload and seeing how much got used. It makes a lot more sense if you know form the outset.
## Changelog
:cl:
qol: Drone dispensers now tell you how much material they need per drone.
/:cl:

### CODE REVIEW

File: code/game/machinery/droneDispenser.dm [hunk 1]
```diff
@@ -125,6 +125,8 @@
 
 /obj/machinery/drone_dispenser/examine(mob/user)
 	. = ..()
+	if (iron_cost > 0 || glass_cost > 0)
+		. += span_notice("It needs [iron_cost] iron and [glass_cost] glass to produce one drone shell.")
```
Comment (JohnFulpWillard) [inline code/game/machinery/droneDispenser.dm:129]: You check if it needs either iron or glass, but then the text implies it will require both?

Comment (NamelessFairy) [reply code/game/machinery/droneDispenser.dm:129]: I'm not actually sure how to code it in a way that makes only 1 material appears. As far as I'm aware . += on examine will always make a new line. I don't want it to say "It needs 1000 iron to produce one drone shell" (newline) "It needs 1000 glass to produce one drone shell"

Comment (NamelessFairy) [reply code/game/machinery/droneDispenser.dm:129]: Nevermind I though or a way

File: code/game/machinery/droneDispenser.dm [hunk 2]
```diff
@@ -125,6 +125,16 @@
 
 /obj/machinery/drone_dispenser/examine(mob/user)
 	. = ..()
+	var/material_requirement_string = "It needs "
+	if (iron_cost > 0)
+		material_requirement_string += "[iron_cost] iron "
+	if (iron_cost > 0 && glass_cost > 0)
+		material_requirement_string += "and "
```
Comment (Time-Green) [inline code/game/machinery/droneDispenser.dm:132]:
```suggestion
	if (iron_cost > 0)
		material_requirement_string += "[iron_cost] iron "
		if (glass_cost > 0)
			material_requirement_string += "and "
```
You don't need to check iron_cost again, just use tabs to bring it in scope of your first check

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Comment by Time-Green** [comment]
Dope

**Pull request merged**